ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -869,100 +874,247 @@ public static BlobProperties deserializeBlobPropertiesRecord(CrcInputStream crcS
+          throw new IllegalArgumentException("Unknown update record type: " + updateRecord.getType());
+      }
+      Crc32 crc = new Crc32();
+      crc.update(outputBuffer.array(), startOffset, getRecordSize(updateRecord.getType()) - Crc_Size);
+      outputBuffer.putLong(crc.getValue());
+    }
+
+    /**
+     * @param crcStream the stream that contains the serialized form of an {@link UpdateRecord} of version 3.
+     * @return the deserialized {@link UpdateRecord}
+     * @throws IOException if there are problems reading from stream.
+     * @throws MessageFormatException if the format of the message is unexpected.
+     */
+    static UpdateRecord deserialize(CrcInputStream crcStream) throws IOException, MessageFormatException {
+      UpdateRecord updateRecord;
+      DataInputStream dataStream = new DataInputStream(crcStream);
+      short accountId = dataStream.readShort();
+      short containerId = dataStream.readShort();
+      long updateTimeInMs = dataStream.readLong();
+      UpdateRecord.Type type = UpdateRecord.Type.values()[dataStream.readShort()];
+      switch (type) {
+        case DELETE:
+          updateRecord = new UpdateRecord(accountId, containerId, updateTimeInMs, getDeleteSubRecord(dataStream));
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown update record type: " + type);
+      }
+      long actualCRC = crcStream.getValue();
+      long expectedCRC = dataStream.readLong();
+      if (actualCRC != expectedCRC) {
+        throw new MessageFormatException(
+            "update record data is corrupt. Expected CRC: " + expectedCRC + ", Actual CRC: " + actualCRC,
+            MessageFormatErrorCodes.Data_Corrupt);
+      }
+      return updateRecord;
+    }
+
+    /**
+     * @param inputStream the stream that contains the serialized form of a {@link DeleteSubRecord}.
+     * @return the deserialized {@link DeleteSubRecord}
+     * @throws IOException if there are problems reading from stream.
+     * @throws MessageFormatException if the format of the message is unexpected.
+     */
+    private static DeleteSubRecord getDeleteSubRecord(DataInputStream inputStream)
+        throws IOException, MessageFormatException {
+      short version = inputStream.readShort();
+      switch (version) {
+        case Delete_Version_V1:
+          return Delete_Sub_Format_V1.deserialize(inputStream);
+        default:
+          throw new MessageFormatException("delete record version not supported: " + version,
+              MessageFormatErrorCodes.Unknown_Format_Version);
+      }
+    }
+  }
+
+  /**
+   *  - - - - - - - - - - - - -
+   * |         |               |
+   * | version |   delete byte |
+   * |(2 bytes)|    (1 byte)   |
+   * |         |               |
+   *  - - - - - - - - - - - - -
+   *  version         - The version of the delete record
+   *
+   *  delete byte     - Takes value 0 or 1. If it is set to 1, it signifies that the blob is deleted. The field
```

Comment:
I would drop the bit about undeletes, as we have not decided on how undeletes will be supported in the future. Based on the information we have today, just the version field would suffice, really. Based on how and when we decide to support undeletes (if at all), we can introduce a new delete version.

Reply:
Fair enough - i copied this from the V1 class that existed and didn't change it. Have removed the line about undeletes